core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -750,6 +752,10 @@ private[spark] class TaskSetManager(
       if (tasksSuccessful == numTasks) {
         isZombie = true
       }
+    } else if (fetchFailedTaskIndexSet.contains(index)) {
```

Comment:
Why are you making this change? I don't quite get it.

Reply:
The change of `fetchFailedTaskIndexSet` is to ignore the task success event if the stage is marked as failed, as Wenchen's suggestion in before comment.

Reply:
We should handle this case in DAGScheduler, then we can look up the stage by task id, and see if the stage is failed or not. Then we don't need `fetchFailedTaskIndexSet`.

Reply:
Great thanks for you two's guidance guidance, that's more clear and the UT add for reproducing this problem can also used for checking it!